### DISCUSSION TITLE
XHR failed

### DISCUSSION THREAD

**Discussion by gerroon** [VSCodium/vscodium#1382] [Q&A]
Hi

I am using  1.74.3 on Windows and when I refresh the extensions this is what I get in the console. How do I properly get extension updates?

```
[Extension Host] joplin folder tree refresh:  19
workbench.desktop.main.js:616   ERR RequestService#request (browser) - error 'https://marketplace.visualstudio.com/_apis/public/gallery'/extensionquery Error: XHR failed
    at b.onerror (workbench.desktop.main.js:72:1428)
workbench.desktop.main.js:774 Error: XHR failed
    at b.onerror (workbench.desktop.main.js:72:1428)
c @ workbench.desktop.main.js:774
(anonymous) @ workbench.desktop.main.js:774
invoke @ workbench.desktop.main.js:85
deliver @ workbench.desktop.main.js:85
fire @ workbench.desktop.main.js:85
addNotification @ workbench.desktop.main.js:774
error @ workbench.desktop.main.js:1753
(anonymous) @ workbench.desktop.main.js:2011
invoke @ workbench.desktop.main.js:85
deliver @ workbench.desktop.main.js:85
fire @ workbench.desktop.main.js:85
run @ workbench.desktop.main.js:85
await in run (async)
onClick @ workbench.desktop.main.js:179
(anonymous) @ workbench.desktop.main.js:179
setTimeout (async)
(anonymous) @ workbench.desktop.main.js:179
workbench.desktop.main.js:72          POST vscode-file://vscode-app/d:/APPS/VsCodium/resources/app/out/vs/code/electron-sandbox/workbench/'https://marketplace.visualstudio.com/_apis/public/gallery'/extensionquery net::ERR_FILE_NOT_FOUND

```

**Comment by ymeine** [comment]
_I know the question is old now, but I'm doing researches on the topic, so I can't leave that unanswered._

First of all, from which output view did you get this error?

Second, the error is weirdly formatted: `'https://marketplace.visualstudio.com/_apis/public/gallery'/extensionquery`. There's a single quote in the middle. Could it be that you improperly configured the URL by adding single quotes, in the wrong place?

**Comment by daiyam** [comment]
The single quotes must have been included in the config to [setup the VS Code Marketplace](https://github.com/VSCodium/vscodium/blob/master/DOCS.md#howto-vscode-marketplace).

**Reply by gerroon** [reply]
Hi, that document does not exist. Do you know if there is a solution to this? I am using 1.85.2, and the issue is still there. I have extensions that I installed before and they can access internet alright.

**Reply by daiyam** [reply]
https://github.com/VSCodium/vscodium/blob/master/docs/index.md#extensions-marketplace

**Comment by gerroon** [comment]
I figured this out. Anyone who has similar issues who ended up trying everything under the sun give it a try. I removed the ENV variables I set, now it works. I totally forgot that it was one of the things I tried back then.

**Comment by daiyam** [comment]
Personally, I don't do that anymore and use https://open-vsx.org/extension/zokugun/vsix-manager
